<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>912</width>
    <height>916</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Widget</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_9">
   <item>
    <widget class="QWidget" name="widget_11" native="true">
     <property name="minimumSize">
      <size>
       <width>530</width>
       <height>762</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>530</width>
       <height>16777215</height>
      </size>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <widget class="QWidget" name="up" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QWidget" name="leftup" native="true">
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QLabel" name="label1">
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>40</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>40</height>
               </size>
              </property>
              <property name="text">
               <string>初始状态</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="State" name="state" native="true">
              <property name="minimumSize">
               <size>
                <width>180</width>
                <height>180</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>180</width>
                <height>180</height>
               </size>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="arrow">
           <property name="minimumSize">
            <size>
             <width>60</width>
             <height>40</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>60</width>
             <height>40</height>
            </size>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="rightup" native="true">
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QLabel" name="label2">
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>40</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>40</height>
               </size>
              </property>
              <property name="text">
               <string>目标状态</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="State" name="goal" native="true">
              <property name="minimumSize">
               <size>
                <width>180</width>
                <height>180</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>180</width>
                <height>180</height>
               </size>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="widget_2" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLineEdit" name="stateInput"/>
         </item>
         <item>
          <widget class="QPushButton" name="stateChange">
           <property name="text">
            <string>改变初始状态</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="goalInput"/>
         </item>
         <item>
          <widget class="QPushButton" name="goalChange">
           <property name="text">
            <string>改变目标状态</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="goBack">
           <property name="text">
            <string>恢复初态</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="widget_10" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <widget class="QRadioButton" name="bfs">
           <property name="text">
            <string>宽度优先</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="dfs">
           <property name="text">
            <string>深度优先</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="astar">
           <property name="text">
            <string>astar</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="bestFirst">
           <property name="text">
            <string>最佳优先</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QStackedWidget" name="stackedWidget">
        <property name="currentIndex">
         <number>2</number>
        </property>
        <widget class="QWidget" name="page">
         <layout class="QHBoxLayout" name="horizontalLayout_11">
          <item>
           <spacer name="horizontalSpacer_5">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="label_5">
            <property name="font">
             <font>
              <family>Agency FB</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>最大节点数：</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="bnum"/>
          </item>
          <item>
           <widget class="QPushButton" name="bNumChange">
            <property name="text">
             <string>确定</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_6">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_2">
         <layout class="QHBoxLayout" name="horizontalLayout_10">
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="label_4">
            <property name="font">
             <font>
              <family>Agency FB</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>最大节点数：</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="dnum"/>
          </item>
          <item>
           <widget class="QPushButton" name="dNumChange">
            <property name="text">
             <string>确定</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_4">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>80</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_3">
         <widget class="QLabel" name="label_6">
          <property name="geometry">
           <rect>
            <x>100</x>
            <y>10</y>
            <width>241</width>
            <height>41</height>
           </rect>
          </property>
          <property name="font">
           <font>
            <family>Agency FB</family>
            <pointsize>16</pointsize>
           </font>
          </property>
          <property name="text">
           <string>g(n):已走的步数</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
         <widget class="QWidget" name="widget_12" native="true">
          <property name="geometry">
           <rect>
            <x>60</x>
            <y>110</y>
            <width>381</width>
            <height>161</height>
           </rect>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <widget class="QRadioButton" name="ah1">
             <property name="text">
              <string>放错的数码的个数</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="ah2">
             <property name="text">
              <string>放错的数码与其正确位置之间的曼哈顿距离</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="ah3">
             <property name="text">
              <string>放错的数码与其正确位置之间的字符串距离</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QLabel" name="label_7">
          <property name="geometry">
           <rect>
            <x>70</x>
            <y>80</y>
            <width>141</width>
            <height>31</height>
           </rect>
          </property>
          <property name="font">
           <font>
            <family>Agency FB</family>
            <pointsize>16</pointsize>
           </font>
          </property>
          <property name="text">
           <string>h(n):</string>
          </property>
         </widget>
        </widget>
        <widget class="QWidget" name="page_4">
         <widget class="QWidget" name="widget_13" native="true">
          <property name="geometry">
           <rect>
            <x>60</x>
            <y>90</y>
            <width>381</width>
            <height>161</height>
           </rect>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_7">
           <item>
            <widget class="QRadioButton" name="bh1">
             <property name="text">
              <string>放错的数码的个数</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="bh2">
             <property name="text">
              <string>放错的数码与其正确位置之间的曼哈顿距离</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="bh3">
             <property name="text">
              <string>放错的数码与其正确位置之间的字符串距离</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QLabel" name="label_8">
          <property name="geometry">
           <rect>
            <x>80</x>
            <y>50</y>
            <width>141</width>
            <height>31</height>
           </rect>
          </property>
          <property name="font">
           <font>
            <family>Agency FB</family>
            <pointsize>16</pointsize>
           </font>
          </property>
          <property name="text">
           <string>h(n):</string>
          </property>
         </widget>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="widget" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QPushButton" name="search">
           <property name="text">
            <string>开始搜索</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="printTree">
           <property name="text">
            <string>输出搜索树</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="printRoad">
           <property name="text">
            <string>输出最短路径</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="widget_9" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <widget class="QWidget" name="widget_7" native="true">
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QWidget" name="widget_3" native="true">
              <layout class="QHBoxLayout" name="horizontalLayout_4">
               <item>
                <widget class="QLabel" name="label">
                 <property name="text">
                  <string>耗费时间：</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="time"/>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QWidget" name="widget_6" native="true">
              <layout class="QHBoxLayout" name="horizontalLayout_6">
               <item>
                <widget class="QLabel" name="label_3">
                 <property name="text">
                  <string>最短路径：</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="steps"/>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="widget_8" native="true">
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QWidget" name="widget_4" native="true">
              <layout class="QHBoxLayout" name="horizontalLayout_5">
               <item>
                <widget class="QLabel" name="label_2">
                 <property name="text">
                  <string>节点个数：</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QWidget" name="widget_5" native="true"/>
               </item>
               <item>
                <widget class="QLineEdit" name="num"/>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="algorithmAna">
              <property name="text">
               <string>算法分析</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QTreeWidget" name="treeWidget">
     <column>
      <property name="text">
       <string>节点</string>
      </property>
     </column>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>State</class>
   <extends>QWidget</extends>
   <header location="global">state.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
